components/pixels/lib.rs

```diff
@@ -257,6 +291,48 @@ fn is_webp(buffer: &[u8]) -> bool {
     buffer[8..].len() >= len && &buffer[8..12] == b"WEBP"
 }
 
+fn gif_decode(buffer: &[u8], cors_status: CorsStatus) -> Option<Image> {
+    let Ok(dec) = GifDecoder::new(Cursor::new(buffer)) else {
+        return None;
+    };
+    let mut width = 0;
+    let mut height = 0;
+    let mut frames = vec![];
+    let frames_result = dec.into_frames();
+    for frame_result in frames_result {
+        let mut frame = match frame_result {
+            Ok(frame) => frame,
+            Err(e) => {
+                debug!("decode GIF frame error: {}", e);
+                break;
```

Comment:
It's a fair point that a a broken frame can just be skipped, so this could simply be a `continue`.